src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractProject_Options.cs

```diff
@@ -80,9 +79,26 @@ protected CommandLineArguments SetArgumentsAndUpdateOptions(string commandLine)
         {
             AssertIsForeground();
 
-            var commandLineArguments = SetArguments(commandLine);
+            var splitArguments = CommandLineParser.SplitCommandLineIntoArguments(commandLine, removeHashComments: false).ToImmutableArray();
+            return SetArgumentsAndUpdateOptions(splitArguments);
+        }
+
+        private CommandLineArguments SetArgumentsAndUpdateOptions(ImmutableArray<string> splitCommandLine)
+        {
+            AssertIsForeground();
+
+            var parsedCommandLineArguments = CommandLineParserService?.Parse(splitCommandLine, this.ContainingDirectoryPathOpt, isInteractive: false, sdkDirectory: System.Runtime.InteropServices.RuntimeEnvironment.GetRuntimeDirectory());
+            SetArgumentsAndUpdateOptions(splitCommandLine, parsedCommandLineArguments);
+            return parsedCommandLineArguments;
+        }
+
+        protected void SetArgumentsAndUpdateOptions(ImmutableArray<string> splitCommandLine, CommandLineArguments commandLineArguments)
+        {
+            AssertIsForeground();
+
+            SetArgumentsCore(splitCommandLine, commandLineArguments);
             UpdateOptions();
-            return commandLineArguments;
+            SetRuleSetFileCore(commandLineArguments?.RuleSetPath);
```

Comment:
So this call goes after UpdateOptions? Just by the names I would have imagined the *Core methods go first, and then UpdateOptions actually reflects the whole thing once we're done.